### ISSUE TITLE
"E488: Traling characters: <NoteType>" On attempting to call :Anki <NoteType>.

### ISSUE THREAD

**Issue by UsuallySleebee** [rareitems/anki.nvim#1]
I'm not 100% sure if I'm following the steps to use this correctly or what but when I try to fill in the template with `Anki: <notetype>` I get given an E488 error.

**My config:**

```lua
local status_ok, anki = pcall(require, "anki")
if not status_ok then
	return
end

anki.setup({
    tex_support = false;
    models = {
        ["C#"] = "Programming::Languages::C#",
        ["Java"] = "Programming::Languages::Java",
        ["DesignPattern"] = "Programming::Software Design::Design Patterns",
        ["FunctionalProgramming"] = "Programming::Software Design::FunctionalProgramming",
        ["OOP"] = "Programming::Software Design::Object Oriented Programming",
        ["Graphics"] = "Programming::Graphics",
    }
})
```

Steps to reproduce:
- Open Neovim.
- Create a Test.anki file.
- Open it and enter the command `:Anki Java`
- Will produce "E488: Trailing characters: Java"

**Comment by rareitems** [comment]
Hi
Do you actually have a 'Java' note type in Anki? You can check your note types inside Anki in `Tools` -> `Manage Note Types`.

I think there is a little bit of misunderstanding here. The "note type" is Anki's thing and not from this plugin. It is a way you can structure your notes (https://docs.ankiweb.net/getting-started.html#note-types).
`models` in config associates a single note type with a single deck.

I think what you are trying to do is to add a single note type to many different decks. You can do that using 'anki.ankiWithDeck' function.
Your config can be empty then
```lua
anki.setup()
```
Assuming you want to use `Basic` (the default Anki notetype)
There is no command for that function so you probably want to define keymaps
```lua
vim.keymap.set("n", "<leader>a1", function()
  anki.ankiWithDeck("Programming::Languages::C#", "Basic")
end)
vim.keymap.set("n", "<leader>a2", function()
  anki.ankiWithDeck("Programming::Languages::Design Pattern", "Basic")
end)
```

Or you can use tags to organize your Anki collection (which is what most people recommend for Anki). Your config then would look something like this then

```lua
{
  models = {
    ["Basic"] = "BasicDeck",
  },
  contexts = {
    ["C#"] = { tags = "Programming::Languages::C#" },
    ["Java"] = { tags = "Programming::Languages::Java" },
    ["DesignPattern"] = { tags = "Programming::SoftwareDesign::DesignPatterns" }, -- spaces are not supported in tags
    ["FunctionalProgramming"] = { tags = "Programming::SoftwareDesign::FunctionalProgramming" },
    ["OOP"] = { tags = "Programming::SoftwareDesign::ObjectOrientedProgramming" }, -- spaces are not supported in tags
    ["Graphics"] = { tags = "Programming::Graphics" },
  },
}
```
All of your cards will be send to one deck, but you can easily find them using tags.
Then you can use `:AnkiSetContext OOP` for example and `:AnkiWithContext` to make a form that will be send with that specific tag.

Also are you sure you didn't get an any errors after opening `Test.anki` file?. You can check with `:messages`
If you have 'noice.nvim' or 'vim.opt.cmdheight' set to '0' they might skip them.

**Comment by UsuallySleebee** [comment]
> Do you actually have a 'Java' note type in Anki? You can check your note types inside Anki in Tools -> Manage Note Types.

Oh, yeah I see where I went wrong now. Sorry that's my bad. For some reason, I thought the notetype you were using in the config was like a "nickname" of sorts for the note you're entering in nvim. Brain not working today, I guess.

I cleared out the other note types (if you have any notes types in the config that aren't in your anki, it fails) and now it's working. Thank you.

> I think what you are trying to do is to add a single note type to many different decks. You can do that using 'anki.ankiWithDeck' function. Your config can be empty then

This is exactly what I want. Legend, thank you.

> Or you can use tags to organize your Anki collection (which is what most people recommend for Anki). Your config then would look something like this then

I use both tags, decks, and profiles to organise my cards. For example, my C# deck will have cards that share a bunch of tags like "Programming", "Programming_Language", "Design_Pattern", "Fundamental_Knowledge", "C#", "Functional_Programming", "Language_Basics", "Types", etc etc.

That way I can review a specific language really easily at a glance when opening the app (and see how many things need to be reviewed) or I can get way more specific and say I want to review design patterns that are in functional programming only. (Mostly good for classes and that.)

> Also are you sure you didn't get an any errors after opening Test.anki file?. You can check with :messages If you have 'noice.nvim' or 'vim.opt.cmdheight' set to '0' they might skip them.

Only the error messages from my misuse of the notetypes.
